Things_You_Should_Know/things_you_should_know.md

```diff
@@ -0,0 +1,400 @@
+
+> **Tip:** 
+> 
+> You can pipe the output of tail to a new file in case you want to download it. Simply append "> filename.txt" to the command above.
+
+### You can discover and download the latest Docker logs using Kudu
+
+|**CODE**|**CONTAINER**|
+|:------:|:-----------:|
+| ⛔    | ✅          |
+
+**App Service** has an API that allows you to easily see the current Docker log details (such as the filename, etc.) and also download the current Docker logs in Zip format.
+
+To see details on the current Docker logs in JSON format, you can use this URL:
+
+`https://<yout-app-name>.scm.azurewebsites.net/api/logs/docker`
+
+You can get to this easily by going to [**Kudu** (Advanced Tools)](https://github.com/projectkudu/kudu/wiki) and then appending "/api/logs/docker" to the URL. The output of this will be a JSON response with the most relevant and current Docker logs.
+
+If you want to download the logs shown in the above API in Zip format, append "zip" to the URL. For example:
+
+`https://<yout-app-name>.scm.azurewebsites.net/api/logs/docker/zip`
+
+
+### You can view the most recent entries in your Docker log in the Azure portal
+
+|**CODE**|**CONTAINER**|
+|:------:|:-----------:|
+| ⛔    | ✅          |
+
+To make it easier to see what's going on with your Docker container, we show you the latest entries in the Docker log in the Azure portal. Simply click on **Container Settings** in the portal menu and you'll see the Docker log.
+
+You can click the Copy button to copy the Docker log. You can also download the full Docker log by clicking on Download Logs.
+
+### If your container takes a long time to start, increase the start time limit
+
+|**CODE**|**CONTAINER**|
+|:------:|:-----------:|
+| ⛔    | ✅          |
+
+**App Service** will wait 230 seconds for a container to start, initialize and respond to an HTTP request. **App Service** interprets this first response to an HTTP request as an idication that your app has initialized and is ready to recive further requests. If the timout of 230 seconds is reached with no response from the app, then it's considerd a failure to start. 
+
+Some containers might need more than 230 seconds to start, the default timeout can be overridden through the `WEBSITES_CONTAINER_START_TIME_LIMIT` app setting. This app setting can be set to a maximum of 1800 seconds.
+
+### Your container must respond to an HTTP ping
+
+|**CODE**|**CONTAINER**|
+|:------:|:-----------:|
+| ⛔    | ✅          |
+
+**App Service** will a certain amount of time for your container to start before we consider it be a failed start. In order for **App Service** to consider a container to be successfully started, the container must start and it must respond to an HTTP ping. If the container starts but does not respond to a ping, we will eventually log an event in the Docker log saying that it didn't start.
```

Comment:
Http ping health check on Linux:
a.	By default, we ping the / of the app and ensure its returning some valid http status code at the start of the container – if the container doesn’t respond within 4 minutes, we stop the container. Note that this specific check is only done during the cold start.   Note - I don't know if this cold start health ping is configurable on our platform.
b.	URL specified by WEBSITE_HEALTH_CHECK_URL – This is configurable by the customer - We will ping this URL and if the container doesn’t respond 200 for ~20 seconds, we will stop the container and restart the site. This specific check is done on a timer thread.